dbt/include/glue/macros/materializations/incremental/incremental.sql

```diff
@@ -61,16 +62,30 @@
         {% endif %}
       {% else %}
         {# /*-- Relation must be merged --*/ #}
-        {%- call statement('create_tmp_view') -%}
-          {{ create_table_as(True, tmp_relation, sql) }}
-        {%- endcall -%}
+        {% if file_format == 'iceberg' and schema_change_mode in ('append_new_columns', 'sync_all_columns') %}
+          {%- call statement('create_tmp_table') -%}
+            create or replace table {{ tmp_relation.include(schema=true) }} using iceberg as {{ sql }}
+          {%- endcall -%}
+          {%- do process_schema_changes(on_schema_change, tmp_relation, target_relation) -%}
+
+          {%- set dest_columns = adapter.get_columns_in_relation(target_relation) -%}
+          {%- set dest_cols_csv = dest_columns | map(attribute='name') | join(', ') -%}
+          {% set build_sql %}
+          insert into {{ target_relation }} ({{ dest_cols_csv }}) select {{ dest_cols_csv }} from {{ tmp_relation.include(schema=true) }}
+          {% endset %}
+
+        {% else %}
+          {%- call statement('create_tmp_relation') -%}
+            create or replace temporary view {{ tmp_relation.include(schema=false) }} as {{ sql }}
```

Comment:
Can we reuse `{{ create_temporary_view(tmp_relation, sql) }}`?

Reply:
Fixed